app/routes.py

```diff
@@ -46,9 +50,13 @@ def create():
 #Submit should always include an id parameter
 @app.route('/submit', methods=['POST'])
 def submit():
+    updateDB()
     submitForm = SubmitForm()
     form = LoginForm()
     challenge = getChallenge(int(submitForm.challenge_id.data)).toDict()
+    if challenge.finished:
+        # There has to be a better way to communicate that the time has run out other than reloading the page
```

Comment:
Agree with comment, but fine with this for now

Reply:
^